Feature/#16 single log line

ruud opened the pull request:
single log submit protocol plus it test

Commit:
setup integration test senario

Commit:
logReceiverPost request added to IntegrationTestBase

ruud committed:
submit single log line

Added protocol and test to logSubmissionTest

TimSoethout commented on src/main/scala/performanceanalysis/logreceiver/LogReceiver.scala:
```diff
@@ -24,8 +30,26 @@ class LogReceiver extends Server  {
+      case LogSubmitted(componentId, logs) =>
+        Future(HttpResponse(status = StatusCodes.Accepted))
+      case LogParserNotFound(componentId) =>
+        ???
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of ??? operator](https://www.codacy.com/app/scala-academy/performance-analysis/file/2462365447/issues/source?bid=3220915&fileBranchId=3255667#l51)

ruud committed:
Changed unittest

ruud committed:
Naming convention single log line

TimSoethout commented on src/main/scala/performanceanalysis/logreceiver/LogReceiver.scala:
```diff
@@ -24,8 +30,26 @@ class LogReceiver extends Server  {
+      case LogSubmitted(componentId, logLine) =>
+        Future(HttpResponse(status = StatusCodes.Accepted))
+      case LogParserNotFound(componentId) =>
+        ???
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of ??? operator](https://www.codacy.com/app/scala-academy/performance-analysis/file/2462588214/issues/source?bid=3220915&fileBranchId=3255667#l51)

ruud replied:
Old operator! To be implemented in an other story

ruud committed:
pathEnd is logs

TimSoethout commented on src/main/scala/performanceanalysis/logreceiver/LogReceiver.scala:
```diff
@@ -24,8 +29,26 @@ class LogReceiver extends Server  {
+      case LogSubmitted(componentId, logLine) =>
+        Future(HttpResponse(status = StatusCodes.Accepted))
+      case LogParserNotFound(componentId) =>
+        ???
```
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of ??? operator](https://www.codacy.com/app/scala-academy/performance-analysis/file/2462660216/issues/source?bid=3220915&fileBranchId=3255667#l50)

ruud replied:
Also an old operator! To be implemented in an other story, I guess

ruud replied:
Also an old operator! To be implemented in an other story, I guess

ruud commented:
Help wanted for codacy issues, as I am going on a holiday

Seetaramayya committed:
- fixed codacy issue
- Added logic for parsing log line using metrics
- added unit test
- modified integration test

Seetaramayya committed:
adjusted for codacy

Seetaramayya committed:
removed duplication

Seetaramayya committed:
added safe guard

geerdink commented on src/main/scala/performanceanalysis/LogParserActor.scala:
```diff
@@ -13,19 +15,35 @@ object LogParserActor {
+      val key = s"${metric.metricKey}-$logLine"
+      val pattern: Regex = metric.regex.r
+      val list = findAllMatchedValues(pattern, logLine)
+      context.become(normal(metrics, parsedLogs.updated(key, list)))
```
Move context.become statement outside of foreach loop

TimSoethout replied:
`parsedLogs.updated(key,list)` wil be overwritten if `metrics` contains multiple items

geerdink commented on src/main/scala/performanceanalysis/LogParserActor.scala:
```diff
@@ -13,19 +15,35 @@ object LogParserActor {
       sender ! Details(metrics)
 
+    case LogSubmitted(_, logLine) =>
+      log.debug("received log line '{}' to be parsed", logLine)
```
change to SubmitLog ??

geerdink commented on src/main/scala/performanceanalysis/administrator/AdministratorActor.scala:
```diff
@@ -60,7 +60,7 @@ class AdministratorActor(logReceiverActor: ActorRef) extends Actor with ActorLog
-  private def handleGetDetails(logParserActors: Map[String, ActorRef], componentId: String, sender: ActorRef) = {
+  private def handleGetMetrics(logParserActors: Map[String, ActorRef], componentId: String, sender: ActorRef) = {
     routeToLogParser(logParserActors, componentId, sender) { ref =>
       log.debug(s"Requesting details from ${ref.path}")
```
use {} instead of s...
